jobs/payment-jobs/tasks/stale_payment_task.py

```diff
@@ -89,3 +97,40 @@ def _delete_marked_payments(cls):
             except BusinessException as err:  # just catch and continue .Don't stop
                 current_app.logger.warn('Error on delete_payment')
                 current_app.logger.warn(err)
+
+    @classmethod
+    def _verify_recent_payments(cls):
+        """Verify recent invoice with PAYBC."""
+        recent_invoices = InvoiceModel.find_recent_cc_invoices(days=2)
+        if len(recent_invoices) == 0:
+            current_app.logger.info(f'Verify Invoice Job Ran at {datetime.datetime.now(tz=datetime.timezone.utc)}.'
+                                    'But No records found!')
+        for invoice in recent_invoices:
+            try:
+                current_app.logger.info(f'Verify Invoice Job found records.Invoice Id: {invoice.id}')
+                paybc_invoice = DirectPayService.query_order_status(invoice, InvoiceReferenceStatus.ACTIVE.value)
+                if paybc_invoice.get('paymentstatus') in STATUS_PAID:
+                    invoice_reference = InvoiceReference.find_active_reference_by_invoice_id(invoice.id)
+                    current_app.logger.debug(f'Getting receipt details {invoice.id}')
+                    # Call PAYBC web service, get access token and use it in get txn call
+                    direct_pay_service = DirectPayService()
+                    access_token = direct_pay_service.get_token().json().get('access_token')
+                    paybc_transaction_url: str = current_app.config.get('PAYBC_DIRECT_PAY_BASE_URL')
+                    paybc_ref_number: str = current_app.config.get('PAYBC_DIRECT_PAY_REF_NUMBER')
+                    paybc_transaction_number = invoice_reference.invoice_number
+                    if not paybc_transaction_number:
```

Comment:
When would this happen?  don't you need to  see if `invoice_reference` exists or not?

Reply:
I removed invoice reference, just directly call TransactionService.update_transaction